### PR TITLE
Convert codebase to ESM and remove babel

### PR DESCRIPTION
**Pull request by anselmbradford**
Converts the codebase to use ESM with one caveat:
 - `jest.mock` does not yet support ESM. Jest provides `jest.unstable_mockModule` (see https://jestjs.io/docs/ecmascript-modules#module-mocking-in-esm). Despite the example shown, I was not able to get the `spyOn` mocks to work in four TDP tests and commented them out for now.  See https://github.com/cfpb/consumerfinance.gov/commit/168644b2e3c662fd94e2955c94a26ac0f77b886d

- ~https://github.com/cfpb/consumerfinance.gov/pull/7403 needs to go in first for unit tests to pass.~

## Removals

- Remove babel.

## Changes

- Convert scripts to ESM modules.

## How to test this PR

1. `yarn build` should pass.
2. `yarn lint` should pass.
3. `yarn jest` should pass (pending FIG PR)

### CODE REVIEW

**Review by wpears — CHANGES_REQUESTED** [review]

**Comment by wpears** [comment]
This is soooooooooooo rad! Good job on all the work that went into this!

File: cfgov/unprocessed/apps/rural-or-underserved-tool/js/call-census.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
-import jsonpP from 'jsonp-p';
+import jsonpPModule from 'jsonp-p';
```
[the remark this answers is not in the document]
Comment (anselmbradford) [reply cfgov/unprocessed/apps/rural-or-underserved-tool/js/call-census.js:1]: I get `Uncaught (in promise) TypeError: (0 , an.default) is not a function`

File: cfgov/unprocessed/apps/rural-or-underserved-tool/js/call-tiger.js [hunk 2]
```diff
@@ -1,4 +1,6 @@
-import jsonpP from 'jsonp-p';
+import jsonpPModule from 'jsonp-p';
```
[the remark this answers is not in the document]
Comment (anselmbradford) [reply cfgov/unprocessed/apps/rural-or-underserved-tool/js/call-tiger.js:1]: I get `Uncaught (in promise) TypeError: (0 , an.default) is not a function`

This whole 6-year old dependency should likely be stripped out and approached a different way.

File: .eslintrc.cjs [hunk 3]
```diff
@@ -1,5 +1,7 @@
 module.exports = {
-  parser: '@babel/eslint-parser',
+  parserOptions: {
+    sourceType: 'module',
```
[the remark this answers is not in the document]
Comment (anselmbradford) [reply .eslintrc.cjs:3]: For now it does unfortunately. The underlying dependency does not yet treat this as ESM. The error is :

```
Error [ERR_REQUIRE_ESM]: require() of ES Module /[user project path]/consumerfinance.gov/.eslintrc.js from /[user project path]/consumerfinance.gov/node_modules/@eslint/eslintrc/dist/eslintrc.cjs not supported.
Instead change the require of .eslintrc.js in /[user project path]/consumerfinance.gov/node_modules/@eslint/eslintrc/dist/eslintrc.cjs to a dynamic import() which is available in all CommonJS modules.
```

File: cfgov/unprocessed/apps/rural-or-underserved-tool/js/call-tiger.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (wpears) [reply cfgov/unprocessed/apps/rural-or-underserved-tool/js/call-tiger.js:1]: Agreed, I might actually put that in as separate PR realllll quick.

**Review by wpears — APPROVED** [review]

**Pull request merged**